tv_apps/smart-system/js/external_storage_monitor.js

```diff
@@ -503,19 +499,13 @@
      * @memberof ExternalStorageMonitor.prototype
      * @param {callback} function The callback will be run while get total space
      */
-    getTotalSpace: function(callback) {
-      var usedSpace, freeSpace;
-      var self = this;
-      this._storage.usedSpace().onsuccess = function(e) {
-        usedSpace = e.target.result;
-        self._storage.freeSpace().onsuccess = function(e) {
-          freeSpace = e.target.result;
-          var totalSpace = self.formatSize(usedSpace + freeSpace);
-          if (callback) {
-            callback(totalSpace);
-          }
-        };
-      };
+    getTotalSpace: function() {
+      return Promise.all([
+        this._storage.usedSpace(),
```

Comment:
https://bugzilla.mozilla.org/show_bug.cgi?id=1084381

It seems that DOMRequest implement's `DOMRequest.prototype.then` making it compatible with Promise.